src/Output/Types.hs

```diff
@@ -332,3 +359,250 @@ writeSignatures store xs = do
         VM.write v i $ fromIntegral $ BS.length b
     v <- V.freeze v
     storeWrite store TypesSigPositions v
+
+readSignatures :: StoreRead -> [Sig Name]
+readSignatures store = go splitters bs
```

Comment:
This appears to read all the signatures into memory - is that right? So you are doing ~5 times (based on the variants) a full scan of every type signature?

Reply:
I think we have to at least read the whole collection of `TypesSigPositions` (which is really more like a vector of *sizes*), and then do some kind of `scan` to get the set of `(offset, size)` pairs.

On a query from the command line, this accounts for ~5% of the overall run time and allocation.

Reply:
At 5% it's hard to bother too much just yet. If we ever want to do better we'll have to be more smart about avoiding the scan through the vectors, and making them actual offsets we can jump to directly.